src/treq/test/test_content.py

```diff
@@ -173,3 +184,21 @@ def test_text_content_default_encoding_no_header(self):
         self.protocol.connectionLost(Failure(ResponseDone()))
 
         self.assertEqual(self.successResultOf(d), u'\xa1')
+
+    def test_text_content_unicode_headers(self):
+        """
+        Header parsing is robust against unicode header names and values.
+        """
+        self.response.headers = Headers({
+            b'Content-Type': [
```

Comment:
Whoa, interesting. I had read [the Headers docstring](https://twistedmatrix.com/documents/17.1.0/api/twisted.web.http_headers.Headers.html#setRawHeaders) and assumed it was based on some spec:

> Most methods accept bytes and unicode, with an internal bytes representation. When passed unicode, header names (e.g. 'Content-Type') are encoded using ISO-8859-1 and header values (e.g. 'text/html;charset=utf-8') are encoded using UTF-8. Some methods that return values will return them in the same type as the name given.

But I don't see any such reference in [#8129](https://twistedmatrix.com/trac/ticket/8129) nor the [merge](https://github.com/twisted/twisted/commit/a1f64685abefed887af611455a5436d7a6fa0dc4).

So really, :man_shrugging:. This PR does fix the Python 2/3 asymmetry, and correct handling of non-UTF-8 JSON. Is there anything left blocking it, or can it be merged?